exasol_integration_test_docker_environment/test/test_populate_data.py

```diff
@@ -0,0 +1,99 @@
+        test_container_content = get_test_container_content((test_container_runtime_mapping,))
+        cls.environment = \
+            cls.test_environment.spawn_docker_test_environment(name=cls.docker_environment_name,
+                                                               test_container_content=test_container_content)
+
+    @classmethod
+    def tearDownClass(cls):
+        utils.close_environments(cls.environment, cls.test_environment)
+
+    def _populate_data(self, reuse=False):
+        task = generate_root_task(task_class=TestDataPopulateData,
+                                  environment_name=self.environment.name,
+                                  db_user=self.environment.db_username,
+                                  db_password=self.environment.db_password,
+                                  reuse_data=reuse,
+                                  bucketfs_write_password=self.environment.bucketfs_password,
+                                  test_environment_info=self.environment.environment_info,
+                                  )
+        try:
+            success = luigi.build([task], workers=1, local_scheduler=True, log_level="INFO")
+            if not success:
+                raise Exception("Task failed")
+        except Exception as e:
+            task.cleanup(False)
+            raise RuntimeError("Error uploading test file.") from e
+
+    def setUp(self) -> None:
+        self._execute_sql_on_db("DROP SCHEMA IF EXISTS TEST CASCADE;")
+
+    def _execute_sql_on_db(self, sql: str) -> str:
+        with ContextDockerClient() as docker_client:
+            test_container = docker_client.containers.get(self.environment.environment_info.
+                                                          test_container_info.container_name)
+            db_info = self.environment.environment_info.database_info
+            db_user_name = self.environment.db_username
+            db_password = self.environment.db_password
+            cmd = f"$EXAPLUS -q -c '{db_info.host}:{db_info.db_port}' " \
+                  f"-u '{db_user_name}' -p '{db_password}' -sql '{sql}' " \
+                  f"-jdbcparam 'validateservercertificate=0'"
+
+            bash_cmd = f"""bash -c "{cmd}" """
+            exit_code, output = test_container.exec_run(cmd=bash_cmd)
+
+        return output.decode("utf-8")
+
+    def test_populate_data(self):
+        self._populate_data()
+        result = self._execute_sql_on_db("SELECT count(*) FROM TEST.ENGINETABLE;")
+        expected_result = '\nCOUNT(*)             \n---------------------\n                  100\n\n'
+        self.assertEquals(result, expected_result)
+
+    def test_populate_data_reuse_data(self):
+        self._populate_data()
+        # Delete a row
+        self._execute_sql_on_db("DELETE FROM TEST.ENGINETABLE WHERE INT_INDEX=1;")
+        # Execute populatedata again. Expectation is that task will be skipped and data remains untouched.
+        self._populate_data(reuse=True)
```

Comment:
I added a new test to validate that.
1. During the test I realized that populateData did not raise an error when Exaplus failed (need to add -x option).
2. And then test_populate_data_reuse_data() started to fail, too, because the db environment had not set the reuse flag